hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDWriteClient.java

```diff
@@ -511,8 +512,13 @@ protected HoodieTable doInitTable(HoodieTableMetaClient metaClient, Option<Strin
    */
   private void initializeMetadataTable(Option<String> inFlightInstantTimestamp) {
     if (config.isMetadataTableEnabled()) {
-      SparkHoodieBackedTableMetadataWriter.create(context.getHadoopConf().get(), config,
+      HoodieTableMetadataWriter writer = SparkHoodieBackedTableMetadataWriter.create(context.getHadoopConf().get(), config,
           context, Option.empty(), inFlightInstantTimestamp);
+      try {
+        writer.close();
```

Comment:
Also, i'm wondering if it would be better to make the metadata writer a member of `BaseHoodieWriteClient` and define the `initializeMetadataTable` method there with concrete implementation in Spark/Flink write client. Then we can close the metadata writer in `BaseHoodieWriteClient#close`.

Reply:
nope, each commit or update has to explicitly instantiate a new writer so that its get all the latest state. thats why we don't have an instance of metadata writer at the class level.